src/virtual_elements.js

```diff
@@ -98,23 +98,38 @@ const elementOpen = function(tag, key, statics, var_args) {
     }
   }
 
-  for (; i < arguments.length; i += 1, j += 1) {
-    attrsArr[j] = arguments[i];
-  }
+  // If we have looped to the end of the attribute pairs in arguments (implied
+  // by attrsChanged being false) but our known cached attrsArr is longer, then
+  // we'll need to setup newAttrs with the attributes for this pass. So send it
+  // down the attrsChanged path. If we didn't make it to the end, then this'll
+  // make no difference since attrsChanged is already true.
+  attrsChanged = attrsChanged || j < attrsArr.length;
+
+  if (attrsChanged) {
+    i -= ~i & 1;
```

Comment:
So purely to golf, can you think of a bitwise equivalent (no `-=`)? Best I could do is `i = i ^ (~(-(i & -i)) | ((i & -i) & ~1))`.